topics/statistics/tutorials/text-mining_simtext/tutorial.md

```diff
@@ -0,0 +1,138 @@
+>
+{: .details}
+
+> ### {% icon hands_on %} Hands-on: Step 1: PubMed query tool
+>
+> 1. Run {% tool [PubMed query](toolshed.g2.bx.psu.edu/repos/iuc/pubmed_by_queries/pubmed_by_queries/0.0.2) %} with the following parameters:
+>    - {% icon param-file %} *"Input file with query terms"*: Input dataset
+>    - *"Number of PMIDs (or abstracts) to save per ID"*: `500`
+>
+>    > ### {% icon comment %} Comment
+>    >
+>    > The tool is also able to save the abstracts as text instead of their PMIDs. This feature is used for another type of analysis (see {% cite gramm2020 %}), or can be used if the user wants to use the tool independent from a workflow to retrieve many abstract texts at once. For the next step of our example in this tutorial, PMIDs are required.
+>    {: .comment}
+>
+{: .hands_on}
+
+# Extraction of biomedical terms from PubMed abstracts
+
+Next, we extract the 100 most frequent 'Disease' and 'Gene' terms (PubTator annotations) from the PubMed data. All genes with their 100 associated terms are then combined in one large binary matrix. Each row represents a gene and each column one of the extracted terms. This matrix is later used to find similar genes, i.e. genes that have many common terms associated with them.
+
+> ### {% icon hands_on %} Hands-on: Extraction of PubTator annotations
+>
+> 1. Run {% tool [PMIDs to PubTator](toolshed.g2.bx.psu.edu/repos/iuc/pmids_to_pubtator_matrix/pmids_to_pubtator_matrix/0.0.2) %} with the following parameters:
+>    - {% icon param-file %} *"Input file with PMID IDs"*: output of **PubMed query** {% icon tool %}
+>    - *"categories"*: `Genes Diseases`
+>    - *"Number of most frequent terms/IDs to extract."*: `100`
+>
+>    > ### {% icon comment %} PubTator
+>    > PubTator annotates terms of the following categories: Gene, Disease, Mutation, Species and Chemical. 
+>    > In this example we chose to only extract gene and disease terms but you can also select other categories if you are interested in those.
+>    > 
+>    {: .comment}
+>
+{: .hands_on}
+
+
+# Exploration of data in interactive tool
+
+After generating the large binary matrix, we can explore the similarities/ the grouping among the genes in the interactive SimText tool.
+The following features are generated:
+
+1. Word clouds for each gene
+2. Dimension reduction and hierarchical clustering of the binary matrix
+3. Calculation of the adjusted rand index (similarity between text-based grouping and the pre-existing disorder categories)
+4. Table with terms and their frequency among the genes
+
+> ### {% icon hands_on %} Hands-on: Explore data interactively
+>
+> 1. Run {% tool [interactive_tool_simtext_app](interactive_tool_simtext_app) %} with the following parameters:
+>    - {% icon param-file %} *"Input file"*: initial input file with genes and pre-existing grouping
+>    - {% icon param-file %} *"Matrix file"*: output of **PMIDs to PubTator** {% icon tool %}
+>
+> 2. Open interactive tool 
```

Comment:
Ah nice, you found the new snippet syntax :)

This step is probably pretty useful in more tutorials. Would you like to create a snippet (faq) for it? You can pass variables to snippets too likes so:

```
{% snippet faqs/galaxy/interactive_tools_open.md  tool="SimText App" %}
```
and in the snippet itself:

```
..
- Go to User > Active InteractiveTools
- Click on {% include.tool %}
..
```

the tutorial looks really good, thanks @dlal-group :+1:

Reply:
Hi @shiltemann and thanks!

I tried to create and integrate the faq by creating a md file in faqs/galaxy and adding the faq to the tutorial md file. However, I get an error when trying to load the page with jekyll. 

The error messages:

When I use the {% include.tool %} strategy you showed me, using tool="SimText app" or "tool="interactive_tool_simtext_app" I get this error:
`Liquid Exception: Could not locate the included file '.tool' in any of ["…../training-material/"]. Ensure it exists in one of those directories and, if it is a symlink, does not point outside your site source. in faqs/galaxy/interactive_tools_open.md`

When I tried to use the faq without {% include.tool %}  and "tool=.." I get this error:
`Liquid Exception: (<unknown>): could not find expected ':' while scanning a simple key at line 6 column 1 in topics/statistics/tutorials/text-mining_simtext/tutorial.md
`

Could it be that another file is needed? I can see for example that all faqs also have an html file under _site/training-material/faqs.

Thank you for your help!

Reply:
Try `{{ include.tool }}` and that will probably help :)

Reply:
ah, my bad, I messed up the example, yes, the double braces like helena points out should work

Reply:
thank you, it works!:)

Reply:
@hexylena @shiltemann I am submitting a review of a paper on Wednesday and would love to mention that there is a tutorial available. Is there anything else I should change? And are we waiting for other reviewers or is one of you able to merge this PR? I appreciate your help, thank you!

Reply:
I will take a look today!

Reply:
Thank you @bgruening !